### PR TITLE
esri login workflow

### LINKED ISSUE #2437: Login stage needs to save user's esri groups in the session
After a user is authenticated by Esri portal, magda needs to use the access token to query Esri on user's groups and save the groups in the current session.

Magda authz will use the groups info to query OPA. Details follow.

1. At login, get the list of groups that user belongs to 
2. Save this info to session
3. When auth-api proxy opa request, make sure the `input.user.groups` object contains this info from session (Currently, auth-api will auto re-create `input.user` only from session)

It needs to be saved in the passport js session (https://www.airpair.com/express/posts/expressjs-and-passportjs-sessions-deep-dive) then put into the JWT token somehow so that the auth API can pull it out.

### PR DESCRIPTION
**Pull request by mwu2018**
### What this PR does

Fixes #2437  and #2438 
This PR will apply either esri-groups-based or hierarchical-organization-based (default) access control policy to registry api read-only operations,  depending on the deployment configuration property `opa.recordPolicyId`:
1. For the esri policy, a JWT token must claim user's ID and session. 
2. For the default policy, a JWT token only needs to claim user's ID.

To simplify the implementation, this PR also creates two view tables `private_records` and `public_records` where the latter references the former. Records in `public_records` do not have access control and are for public access.

The authz behaviours can be verified by two test suites: `opa/RecordsWithOwnerOrgUnitsOpaPoliciesSpec` and `opa/RecordsWithEsriOpaPoliciesSpec`. Please see `opa/README.md` for details.

This PR uses a generic approach letting JWT token carry extra info required by a specific OPA policy. The JWT will have a generic claim "session" that may contain any extra info required by OPA.
When using Esri OPA policy, the JWT "session" will have a key "esriGroups" to an array of group IDs that a user have access to.

Note:
JWT token is subject to max JWT size limitation. In Esri usecase, some user might have very large number of access groups. If JWT token size is an issue, we need to revise the current approach.

A new deployment file "minikube-esri-dev.yml" is added for local esri deployment.

### Checklist

-   [x] There are unit tests to verify my changes are correct or unit tests aren't applicable
-   [x] I've updated CHANGES.md with what I changed.
-   [x] I've linked this PR to an issue in ZenHub (core dev team only)

### CODE REVIEW

**Comment by AlexGilleran** [comment]
It's not a big deal yet but we should start thinking about how multiple ESRI auth providers work in the same magda instance too 👍

File: magda-gateway/src/createApiRouter.ts [hunk 1]
```diff
@@ -35,9 +35,12 @@ export default function createApiRouter(options: ApiRouterOptions): Router {
 
     proxy.on("proxyReq", (proxyReq, req: any, res, options) => {
         if (jwtSecret && req.user) {
+            console.log(
```
[the remark this answers is not in the document]
Comment (mwu2018) [reply magda-gateway/src/createApiRouter.ts]: Yes. It is a debug left-over

File: magda-typescript-common/src/session/GetUserSession.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            return Maybe.just(session);
+        } catch (e) {
+            return Maybe.nothing<{}>();
+        }
+    } else {
+        return Maybe.nothing<{}>();
+    }
+}
+
+export function getUserGroupsHandling(
```
[the remark this answers is not in the document]
Comment (mwu2018) [reply magda-typescript-common/src/session/GetUserSession.ts]: This is not being used and can be removed.

**Comment by mwu2018** [comment]
> It's not a big deal yet but we should start thinking about how multiple ESRI auth providers work in the same magda instance too 👍

It those use cases I think we need to add policy id columns to records table.

**Pull request merged**